### ISSUE TITLE
Hide `go`

### ISSUE THREAD

**Issue by paldepind** [Funkia/funnel#4]
Hide the `go` function so that generator functions can optionally be supplied directly.

**Comment by limemloh** [comment]
Yes, I agree that this would be a nice feature.

Also have you seen the [`fgo`](https://github.com/Funkia/jabz/blob/master/src/monad.ts#L63) function from [jabz](https://github.com/Funkia/jabz)? It gives almost the same effect:

``` typescript
const main = component<ToView, ViewOut, {}>(
  fgo(function*({emailB, calcLength}): Iterator<Now<any>> {
    const validB = emailB.map(isValidEmail);
    const lengthUpdate = snapshot(emailB, calcLength).map(getLength);
    const lengthB = stepper(0, lengthUpdate);
    return Now.of([{validB, lengthB}, {}]);
  }),
  fgo(function*({validB, lengthB}): Iterator<Component<any>> {
    yield span("Please enter an email address: ");
    const {inputValue: emailB} = yield input();
    yield br;
    yield text("The address is ");
    yield text(validB.map(t => t ? "valid" : "invalid"));
    yield br;
    const {click: calcLength} = yield button("Calculate length");
    yield span(" The length of the email is ");
    yield text(lengthB);
    return Component.of({emailB, calcLength});
  })
);
```

**Comment by paldepind** [comment]
Closed by #7.
